lib/matplotlib/_pylab_helpers.py

```diff
@@ -47,6 +59,54 @@ def get_fig_manager(cls, num):
         return manager
 
     @classmethod
+    def show_all(cls, block=None):
+        """
+        Show all figures.  If *block* is not None, then
+        it is a boolean that overrides all other factors
+        determining whether show blocks by calling mainloop().
+        The other factors are:
+        it does not block if run inside ipython's "%pylab" mode
+        it does not block in interactive mode.
+        """
+
+        managers = cls.get_all_fig_managers()
+        if not managers:
+            return
+
+        for manager in managers:
+            manager.show()
+
+        if block is True:
+            manager._mainloop()
```

Comment:
Yes, because calling mainloop will pause execution here... we only want one to pause it once we have shown all of the managers.